core/transaction/validator.go

```diff
@@ -320,6 +321,14 @@ func CheckTransactionPayload(txn *Transaction) error {
 			return errors.New("asset is not enough")
 		}
 	case *payload.RegisterName:
+		match, err := regexp.MatchString("([a-z]{8,12})", pld.Name)
```

Comment:
Why don't we allow A-Z and shorter/longer name?

Reply:
Same reason why we don't allow several names to one public key :) A-Z would mean than names are case sensitive (it's a big decision) and min/max length chosen are just reasonable defaults that can be changed later.